### PR TITLE
Support configuring etcd flags

### PR DESCRIPTION
**Pull request by DanielXiao**
Add Cluster variable etcdExtraArgs to set arbitrary etcd flags. Add Cluster variable tlsCipherSuites to store the default cipher suites. When etcdExtraArgs does not contain cipher-suites, the default value is set.

Add variable ETCD_EXTRA_ARGS for UX backward compatibility.

### What this PR does / why we need it
We need to support configuring etcd flags since Nimbus testbeds have only low performance storages.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Use the variable ETCD_EXTRA_ARGS to configure etcd flags. See the following example:
ETCD_EXTRA_ARGS: "heartbeat-interval=300;election-timeout=2000"
```

### CODE REVIEW

File: packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml [hunk 1]
```diff
@@ -45,6 +45,20 @@ spec:
...
       namespace: #@ data.values.NAMESPACE
   variables:
+  - name: etcdExtraArgs
+    required: false
+    schema:
+      openAPIV3Schema:
+        type: object
+        additionalProperties:
+          type: string
+        default: {}
```
Comment (wjun) [inline packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml:55]: Is that possible to set default value as {cipher-suites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384}, so you can remove the tlsCipherSuites variable?

File: providers/yttcc/lib/helpers.star [hunk 2]
```diff
@@ -323,15 +323,25 @@ end
...
 # get_labels_map_from_string constructs a map from given string of the format "key1=label1,key2=label2"
 def get_labels_map_from_string(labelString):
-   labelMap = {}
-   for val in labelString.split(','):
+   return get_map_from_string(labelString, ',')
+end
+
+# get_extra_args_map_from_string constructs a map from given string of the format "key1=label1,key2=label2"
+def get_extra_args_map_from_string(labelString):
+   return get_map_from_string(labelString, ';')
```
Comment (wjun) [inline providers/yttcc/lib/helpers.star:331]: Why not use ',' as the delimiter?

File: providers/config_default.yaml [hunk 3]
```diff
@@ -62,6 +62,11 @@ ENABLE_TKGS_ON_VSPHERE7:
 #! Valid values for BUILD_EDITION are 'tce' and 'tkg'.
 BUILD_EDITION:
 
+#! ETCD_EXTRA_ARGS specified etcd flags.
+#! It should be of the format "key1=value1;key2=value2".
```
Comment (wjun) [inline providers/config_default.yaml:66]: Why not use ',' as the delimiter?

**Review by tenczar — APPROVED** [review]
it seems reasonable to me. I'm becoming a little concerned with how much customization is being baked directly into every clusterclass. I think podsecuritypolicy added clusterclass customizations in an interesting way, or maybe we even adopt kustomize in the future...

File: packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml (hunk 1 above)
Comment (DanielXiao) [reply packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml:55]: Exactly. Users must specify cipher-suites in the variable etcdExtraArgs if we don't append the default value. E.g: If users only specify the following args, cipher-suites will be missed.
```
    - name: etcdExtraArgs
      value:
        election-timeout: "2000"
        heartbeat-interval: "300"
```

Comment (DanielXiao) [reply packages/tkg-clusterclass-aws/bundle/config/upstream/base.yaml:55]: variable tlsCipherSuites will be shared as the default cipher suites by all components: apiserver, kube-controller-mananger, kubelet etc. I just don't have time to change them.

File: providers/yttcc/lib/helpers.star (hunk 2 above)
Comment (DanielXiao) [reply providers/yttcc/lib/helpers.star:331]: As you can see, ',' is already used as the delimiter of some flags, e.g. cipher-suites `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384`

File: providers/config_default.yaml (hunk 3 above)
Comment (DanielXiao) [reply providers/config_default.yaml:66]: Same as above.

**Comment by DanielXiao** [comment]
@tenczar @wjun As you can see jsonpatch is aweful, other folks can introduce a patch to change the same fields by accident and there is no way to write unit test to ensure the expected behavior. We should try runtime sdk next release so we can easily modify map, array and leverage unit tests to avoid regression.

**Review by wjun — APPROVED** [review]
LGTM

**Pull request merged**